DEVELOPMENT.md

```diff
@@ -88,6 +96,32 @@ registry which if you are developing locally you could use `export KO_DOCKER_REP
 to use the local one on kind. Please see official
 [KO documentation](https://github.com/google/ko#local-publishing-options) for more information.
 
+#### Running end-to-end tests locally
```

Comment:
First of all, thank you SO much for adding instructions for how to run the tests!


Reading this PR reminded me that this is a file that can use some fixing. Example, that requirement section could go away. For now, my suggestion is that if we group all the testing info together it would make the doc better and it would make the instructions you are adding easier to follow (imo). So, see what happens if you ove a few things around.


First, a note: this PR indicates that there are 2 ways to run the e2e tests, locally vs manually. Once one wraps her head around it, it makes sense. But you will note that I changed that.


Reasoning: because this is a development setup, all tests are to be trigered locally. We don't need to mention that. Also. the "locally" in this PR seems to be the equivalent of 'using scripts to provision a cluster and run all the things", while "manually" seems to mean no setup and scripts needed when there is already a cluster.


So, I changed it to: s/locally/with Kind and s/manually/against a provider.


Maybe it should be s/against a provider/against an existing cluster?


My change iin the naming of those sections might not be super accurate, but I think it's closer in meaning. If it's totally off, feel free to ignore. Or adapt.


1) move the section `### Local Development` to the very end of this file.


2) move the section you added, `#### Running end-to-end tests locally` to under that.


3) move the other section you added, `### Running the e2e tests manually`, to under those two, making it the last section (for now).


Now, let's fix those sections (feel free to use better words, I'm mainly focusing on re-arranging the sections):


- rename `#### Running end-to-end tests locally` to: ### e2e testing`


- add a section: #### Prerequisites
  - move all of the requirements you added to this section and/or subsections below, as appropriate


- add a section: #### Run tests against a provider
  - move the text from your added section `### Running the e2e tests manually` here


- add a section: #### Run tests with Kind
  - move the text from your added section `#### Running end-to-end tests locally` here and/or subsections below, as appropriate


1) Linux


2) MacOS